### PR TITLE
FHIR-45980 Add indication of suitable name usage (Name Context)

### PR DESCRIPTION
**Pull request by brettesler-ext**
Approved to make PR https://confluence.hl7.org/display/HAFWG/2024-06-19+FHIRWG+Meeting+Minutes
added:
Name Context Extension
Name Context ValueSet / CodeSystem
and examples

Output from FSH content- see branch https://github.com/hl7au/au-fhir-base/tree/individual-name-context

### CODE REVIEW

**Comment by brettesler-ext** [comment]
JIRA https://jira.hl7.org/browse/FHIR-45980

File: input/resources/CodeSystem-name-context.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+<CodeSystem xmlns="http://hl7.org/fhir">
+  <id value="name-context"/>
+  <meta>
+    <profile value="http://hl7.org/fhir/StructureDefinition/shareablecodesystem"/>
+  </meta>
+  <extension url="http://hl7.org/fhir/StructureDefinition/structuredefinition-fmm">
+    <valueInteger value="0"/>
+  </extension>
+  <url value="http://terminology.hl7.org.au/CodeSystem/name-context"/>
+  <name value="NameContextCodeSystem"/>
```
Comment (dbojicic) [inline input/resources/CodeSystem-name-context.xml:11]: Remove CodeSystem from the name

File: input/resources/CodeSystem-name-context.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  <title value="Name Context Codes"/>
+  <status value="active"/>
+  <description value="The Name Context code system defines concepts describing usage of names. This provides well known concepts that can be used to indicate usage to achieve an associated purpose e.g. identifier validation or (in)appropriate cultural, religious or social usage of names."/>
+  <content value="complete"/>
+  <concept>
+    <code value="ihi-validation"/>
+    <display value="IHI Validation"/>
+    <definition value="Indicates associated name is for IHI (Individual Healthcare Identifier) Validation"/>
```
Comment (dbojicic) [inline input/resources/CodeSystem-name-context.xml:19]: Change 'Validation' to lower case 'validation' and add a full stop.

File: input/resources/CodeSystem-name-context.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  </concept>
+  <concept>
+    <code value="hpii-validation"/>
+    <display value="HPI-I Validation"/>
+    <definition value="Indicates associated name is for HPI-I (Healthcare Provider Identifier - Individual) Validation"/>
```
Comment (dbojicic) [inline input/resources/CodeSystem-name-context.xml:24]: Change 'Validation' to lower case 'validation' and add a full stop.

File: input/resources/StructureDefinition-name-context.xml [hunk 4]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<StructureDefinition xmlns="http://hl7.org/fhir">
+  <id value="name-context"/>
+  <extension url="http://hl7.org/fhir/StructureDefinition/structuredefinition-fmm">
+    <valueInteger value="0"/>
+  </extension>
+  <url value="http://hl7.org.au/fhir/StructureDefinition/name-context"/>
+  <name value="NameContext"/>
+  <title value="Name Context"/>
+  <status value="active"/>
```
Comment (dbojicic) [inline input/resources/StructureDefinition-name-context.xml:10]: status should be draft

File: input/resources/StructureDefinition-name-context.xml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  <description value="A suitable context of use for a patient, person, practitioner or related person name can be indicated using this extension. &#xA;This includes an ability to provide a coded concept or text description of a suitable use of the associated name.&#xA;The usage described may be varied in nature including suitability for a particular purpose e.g. IHI validation, or highlighting (in)appropriate&#xA;use when referring to an individual e.g. deceased Aboriginal name, transgender deadname."/>
```
Comment (dbojicic) [inline input/resources/StructureDefinition-name-context.xml:11]: Should be in format: "This extension applies to the <resource/element/datatype> and is used to indicate/represent .."

File: input/resources/StructureDefinition-name-context.xml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    <type value="element"/>
+    <expression value="Practitioner.name"/>
+  </context>
+  <type value="Extension"/>
+  <baseDefinition value="http://hl7.org/fhir/StructureDefinition/Extension"/>
+  <derivation value="constraint"/>
+  <differential>
+    <element id="Extension">
+      <path value="Extension"/>
+      <short value="Name Context"/>
```
Comment (dbojicic) [inline input/resources/StructureDefinition-name-context.xml:36]: Should be "Name context"

File: input/resources/CodeSystem-name-context.xml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  <meta>
+    <profile value="http://hl7.org/fhir/StructureDefinition/shareablecodesystem"/>
+  </meta>
+  <extension url="http://hl7.org/fhir/StructureDefinition/structuredefinition-fmm">
+    <valueInteger value="0"/>
+  </extension>
+  <url value="http://terminology.hl7.org.au/CodeSystem/name-context"/>
+  <name value="NameContextCodeSystem"/>
+  <title value="Name Context Codes"/>
+  <status value="active"/>
```
Comment (dbojicic) [inline input/resources/CodeSystem-name-context.xml:13]: status should be draft

File: input/resources/ValueSet-name-context.xml [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  <meta>
+    <profile value="http://hl7.org/fhir/StructureDefinition/shareablevalueset"/>
+  </meta>
+  <extension url="http://hl7.org/fhir/StructureDefinition/structuredefinition-fmm">
+    <valueInteger value="0"/>
+  </extension>
+  <url value="http://terminology.hl7.org.au/ValueSet/name-context"/>
+  <name value="NameContextValueSet"/>
+  <title value="Name Context"/>
+  <status value="active"/>
```
Comment (dbojicic) [inline input/resources/ValueSet-name-context.xml:13]: status should be draft

File: input/resources/CodeSystem-name-context.xml [hunk 9]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CodeSystem xmlns="http://hl7.org/fhir">
```
Comment (dbojicic) [inline input/resources/CodeSystem-name-context.xml:2]: Should have a registered OID as an identifier, see https://github.com/hl7au/au-fhir-base/wiki#terminology-resource-conventions

File: input/resources/ValueSet-name-context.xml [hunk 10]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dbojicic) [inline input/resources/ValueSet-name-context.xml:1]: Should have a registered OID as an identifier, see https://github.com/hl7au/au-fhir-base/wiki#terminology-resource-conventions

**Review by dbojicic — COMMENTED** [review]
There are QA Report error associated with the CodeSystem-name-context & ValueSet-name-context  URL and the IG canonical URL mismatch. 
Include URL as a special-url in ig.xml to eliminate the error.

File: input/resources/CodeSystem-name-context.xml
Comment (dbojicic) [inline input/resources/CodeSystem-name-context.xml:1]: Please include experimental to resolve QA error: 
CodeSystem.experimental: minimum required = 1, but only found 0 (from http://hl7.org/fhir/StructureDefinition/shareablecodesystem|4.0.1)

File: input/resources/ValueSet-name-context.xml
Comment (dbojicic) [inline input/resources/ValueSet-name-context.xml:1]: Please include experimental, to resolve QA error:
ValueSet.experimental: minimum required = 1, but only found 0 (from http://hl7.org/fhir/StructureDefinition/shareablevalueset|4.0.1)

File: input/resources/Practitioner-example-practitioner-name-context.xml [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+<Practitioner xmlns="http://hl7.org/fhir">
+  <id value="example-practitioner-name-context"/>
+  <active value="true"/>
+  <name>
+    <extension url="http://hl7.org.au/fhir/StructureDefinition/name-context">
+      <valueCodeableConcept>
+        <coding>
+          <system value="http://terminology.hl7.org.au/CodeSystem/name-context"/>
+          <code value="hpii-validation"/>
+          <display value="IHI Validation"/>
```
Comment (dbojicic) [inline input/resources/Practitioner-example-practitioner-name-context.xml:11]: Display should be "HPI-I Validation".

File: input/resources/CodeSystem-name-context.xml (hunk 7 above)
Comment (mjosborne1) [reply input/resources/CodeSystem-name-context.xml:13]: I think the word "Codes" is redundant in a CodeSystem resource.

**Review by mjosborne1 — COMMENTED** [review]
Agree with all the changes that @dbojicic has suggested, and you may consider removing the word "Codes" from the CodeSystem as per my comment.  
The examples look fine.

**Review by dbojicic — CHANGES_REQUESTED** [review]
A few changes requested, see inline comments.

File: input/resources/StructureDefinition-name-context.xml (hunk 5 above)
Comment (brettesler-ext) [reply input/resources/StructureDefinition-name-context.xml:11]: reworded as HumanName datatype extension and mentioning the resource type content; similar to medication-type

File: input/resources/CodeSystem-name-context.xml [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  </meta>
+  <extension url="http://hl7.org/fhir/StructureDefinition/structuredefinition-fmm">
+    <valueInteger value="0"/>
+  </extension>
+  <url value="http://terminology.hl7.org.au/CodeSystem/name-context"/>
+  <identifier>
+    <system value="urn:ietf:rfc:3986"/>
+    <value value="urn:oid:2.16.840.1.113883.2.3.4.1.4.31"/>
+  </identifier>
+  <name value="NameContextCodes"/>
```
Comment (dbojicic) [inline input/resources/CodeSystem-name-context.xml:15]: Still contains 'Codes'

File: input/resources/ValueSet-name-context.xml [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+  <url value="http://terminology.hl7.org.au/ValueSet/name-context"/>
+  <identifier>
+    <system value="urn:ietf:rfc:3986"/>
+    <value value="urn:oid:2.16.840.1.113883.2.3.4.2.4.34"/>
+  </identifier>
+  <name value="NameContext"/>
+  <title value="Name Context"/>
+  <status value="draft"/>
+  <experimental value="false"/>
+  <description value="The Name Context value set is used to classify suitable uses or purposes of names."/>
```
Comment (dbojicic) [inline input/resources/ValueSet-name-context.xml:19]: Consider changing 'is used' to 'may be used' to ensure consistency with other value set descriptions.

**Review by dbojicic — CHANGES_REQUESTED** [review]
Two minor changes left.

**Comment by brettesler-ext** [comment]
thanks!

**Comment by brettesler-ext** [comment]
change log added

File: input/resources/CodeSystem-name-context.xml (hunk 12 above)
Comment (dbojicic) [reply input/resources/CodeSystem-name-context.xml]: Fix confirmed.

File: input/resources/ValueSet-name-context.xml (hunk 13 above)
Comment (dbojicic) [reply input/resources/ValueSet-name-context.xml]: Confirmed fix.

**Review by dbojicic — APPROVED** [review]

**Review by mjosborne1 — APPROVED** [review]

**Pull request merged**